tests/test_request_fingerprinter.py

```diff
@@ -0,0 +1,291 @@
+    )
+
+    crawler = get_crawler()
+    no_job_fingerprinter = create_instance(
+        ScrapyZyteAPIRequestFingerprinter, settings=crawler.settings, crawler=crawler
+    )
+
+    request1 = Request("https://example.com", meta={"zyte_api": {"echoData": "foo"}})
+    request2 = Request("https://example.com", meta={"zyte_api": True})
+
+    fingerprint1 = job_fingerprinter.fingerprint(request1)
+    fingerprint2 = no_job_fingerprinter.fingerprint(request2)
+    fingerprint3 = job_fingerprinter.fingerprint(request1)
+    fingerprint4 = no_job_fingerprinter.fingerprint(request2)
+
+    assert fingerprint1 == fingerprint2
+    assert fingerprint3 == fingerprint4
+    assert fingerprint1 == fingerprint3
+
+
+def test_only_end_parameters_matter():
+    """Test that it does not matter how a request comes to use some Zyte API
+    parameters, that the fingerprint is the same if the parameters actually
+    sent to Zyte API are the same."""
+
+    settings = {
+        "ZYTE_API_TRANSPARENT_MODE": True,
+    }
+    crawler = get_crawler(settings_dict=settings)
+    transparent_fingerprinter = create_instance(
+        ScrapyZyteAPIRequestFingerprinter, settings=crawler.settings, crawler=crawler
+    )
+
+    crawler = get_crawler()
+    default_fingerprinter = create_instance(
+        ScrapyZyteAPIRequestFingerprinter, settings=crawler.settings, crawler=crawler
+    )
+
+    request = Request("https://example.com")
+    fingerprint1 = transparent_fingerprinter.fingerprint(request)
+    fingerprint2 = default_fingerprinter.fingerprint(request)
+
+    raw_request = Request(
+        "https://example.com",
+        meta={"zyte_api": {"httpResponseBody": True, "httpResponseHeaders": True}},
+    )
+    fingerprint3 = transparent_fingerprinter.fingerprint(raw_request)
+    fingerprint4 = default_fingerprinter.fingerprint(raw_request)
+
+    auto_request = Request("https://example.com", meta={"zyte_api_automap": True})
+    fingerprint5 = transparent_fingerprinter.fingerprint(auto_request)
+    fingerprint6 = default_fingerprinter.fingerprint(auto_request)
+
+    assert fingerprint1 != fingerprint2
+
+    assert fingerprint3 == fingerprint4
+    assert fingerprint5 == fingerprint6
+
+    assert fingerprint1 == fingerprint3
+    assert fingerprint1 == fingerprint5
+
+
+@pytest.mark.parametrize(
+    "url1,url2,match",
+    (
+        (
+            "https://example.com",
+            "https://example.com",
+            True,
+        ),
+        (
+            "https://example.com",
+            "https://example.com/",
+            True,
+        ),
+        (
+            "https://example.com/?a=1&b=2",
+            "https://example.com/?b=2&a=1",
+            True,
+        ),
+        (
+            "https://example.com/?1",
+            "https://example.com/?2",
+            False,
+        ),
+        (
+            "https://example.com/#1",
+            "https://example.com/#2",
+            True,
```

Comment:
This is actually an interesting point. I can see how it's compatible with the existing approach, and how it works well intransparent mode. But I wonder if we should take fragment in account if browser-related output is requested. Not sure if we can make it a default though, and if it's not a default, it can be introduced later.

Reply:
> But I wonder if we should take fragment in account if browser-related output is requested.

That is a great point. And I think it may be best to address it as part of this pull request.